pkg/util/testappender/internal/dtobuilder/sort.go

```diff
@@ -0,0 +1,73 @@
+package dtobuilder
+
+import (
+	"sort"
+
+	dto "github.com/prometheus/client_model/go"
+)
+
+// sortMetricFamilies sorts metric families, the metrics inside them, and the
+// summaries and histograms inside the individual metrics.
+func sortMetricFamilies(mf []*dto.MetricFamily) {
+	sort.Slice(mf, func(i, j int) bool {
+		return mf[i].GetName() < mf[j].GetName()
+	})
+
+	// Sort the met
+	for _, family := range mf {
+		sortMetrics(family.Metric)
+	}
+}
+
+// sortMetrics sorts a slice of metrics, followed by the quantiles and
+// histogram buckets (if present) inside each metric.
+func sortMetrics(mm []*dto.Metric) {
+	sort.Slice(mm, func(i, j int) bool {
+		return labelsLess(mm[i].GetLabel(), mm[j].GetLabel())
+	})
+
+	for _, m := range mm {
+		if m.Summary != nil {
+			sortQuantiles(m.Summary.GetQuantile())
+		}
+
+		if m.Histogram != nil {
+			sortBuckets(m.Histogram.GetBucket())
+		}
+	}
+}
+
+func labelsLess(a, b []*dto.LabelPair) bool {
```

Comment:
I may not be fully understanding this, but something feels off about the naming. I read this as does A have less labels than B. Where I think its the the inverse.